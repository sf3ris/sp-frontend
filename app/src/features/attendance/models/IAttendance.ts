export interface IAttendance {
    _id: string;
    date: string;
    athletes_ids: string[];
    members_ids: string[];
    title: string;
    type?: string;
}
