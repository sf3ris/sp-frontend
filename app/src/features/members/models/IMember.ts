export interface IMember {
    _id : string;
    name : string;
    last_name : string;
    birth_date : string;
    birth_place : string;
    fiscal_code : string;
    address : string;
    zip_code : string;
    city : string;
    province : string;
    gender: string;
    //gender : keyof typeof Gender;
    phone : string;
    email : string;
}

enum Gender {
    M = "M",
    F = "F"
}