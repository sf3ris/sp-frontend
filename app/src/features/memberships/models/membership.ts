export interface IMembership {
    _id: string;
    start_date : Date;
    end_date : Date;
}