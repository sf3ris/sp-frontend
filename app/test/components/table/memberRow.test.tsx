import React from 'react';
import { shallow } from 'enzyme';
import MemberRowComponent from '../../../src/component/member/Table/memberRow.component';
import { IMember } from '../../../src/features/members/models/IMember';

describe('Member Row Component', () => {

    const onEditFn = jest.fn();
    const onDeleteFn = jest.fn();

    const memberMock : IMember = {
        _id: '123',
        name: 'Mario',
        last_name: 'Rossi',
        birth_date: '1980-10-01',
        birth_place: 'Tolentino',
        fiscal_code: 'MRARSS80E01E783P',
        address: 'Via del lavoro, 1',
        city: 'Tolentino',
        zip_code: '62010',
        province: 'MC',
        gender: 'M',
        phone: '3334885961',
        email: 'mario.rossi@libero.it'
    }

    it('Should match snapshot', () => {

        const wrapper = shallow(
            <MemberRowComponent
                member={memberMock}
                onDeleteClick={onDeleteFn}
                onEditClick={onEditFn} />
        )

        expect(wrapper.debug()).toMatchSnapshot();

    })

    it('Should trigger edit', () => {

        const wrapper = shallow(
            <MemberRowComponent
                member={memberMock}
                onDeleteClick={onDeleteFn}
                onEditClick={onEditFn} />
        )

        const editButton = wrapper.find('#editButtonId_123');
        expect(editButton.length).toBe(1);

        editButton.simulate('click');

        expect(onEditFn).toBeCalled();

    });
    
})
