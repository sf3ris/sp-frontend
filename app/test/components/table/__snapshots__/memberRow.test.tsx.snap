// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Row Component Should match snapshot 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(TableRow))>
    <WithStyles(ForwardRef(TableCell))>
      Mario
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      Rossi
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      01/10/1980
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      MRARSS80E01E783P
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      Tolentino
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      MC
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      mario.rossi@libero.it
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell)) colSpan={2}>
      <WithStyles(ForwardRef(IconButton)) id=\\"editButtonId_123\\" onClick={[Function: onClick]} size=\\"small\\">
        <FontAwesomeIcon icon={{...}} color=\\"green\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton)) id=\\"deleteButtonId_123\\" onClick={[Function: onDelete]} size=\\"small\\">
        <FontAwesomeIcon icon={{...}} color=\\"red\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <ConfirmDialog title=\\"Confermi cancellazione?\\" onOk={[Function: confirmDelete]} onDiscard={[Function: onDiscard]} isOpen={false} />
</Fragment>"
`;
